01/app.ts

```diff
@@ -0,0 +1,44 @@
+class User {
+    private _email: string;
+    private _password: string = '';
+
+    constructor(email: string, password: string) {
+        this._email = email;
+        hashPassword(password).then(hashedPassword => {
+            this._password = hashedPassword;
+        });
+    }
+
+    set email(value: string) {
+        this._email = value;
+    }
+
+    get email(): string {
+        return this._email;
+    }
+
+    async check(email: string, password: string): Promise<boolean> {
+        const hashedPassword = await hashPassword(password); 
+
+        return this._email === email && this._password === hashedPassword; 
+    }
+}
+
+async function hashPassword(password: string): Promise<string> {
```

Comment:
Osobiście dodałbym tą funkcję do wnętrza klasy i za każdym razem kiedy zapisywane jest hasło to ją wykorzystywał - tutaj dodałbym `set` i `get` dla password jak to zrobiłaś dla email.